hooks/pre-command

```diff
@@ -22,12 +29,20 @@ if [[ -n "${BUILDKITE_PLUGIN_GCP_WORKLOAD_IDENTITY_FEDERATION_RENDER_COMMAND:-}"
   BUILDKITE_PLUGIN_GCP_WORKLOAD_IDENTITY_FEDERATION_SERVICE_ACCOUNT="$(echo "${BUILDKITE_PLUGIN_GCP_WORKLOAD_IDENTITY_FEDERATION_SERVICE_ACCOUNT}" | ${BUILDKITE_PLUGIN_GCP_WORKLOAD_IDENTITY_FEDERATION_RENDER_COMMAND})"
 fi
 
+# Parse the list of optional claims to pass to the token request command
+while read -r line ; do
+  [[ -n "$line" ]] && oidc_params+=("--claim" "$line")
+done <<< "$(plugin_read_list OPTIONAL_CLAIMS)"
+
 # Create a temporary directory with both BSD and GNU mktemp
 TMPDIR=$(mktemp -d 2>/dev/null || mktemp -d -t 'buildkiteXXXX')
 
 echo "~~~ :buildkite: Requesting OIDC token from Buildkite"
 
-buildkite-agent oidc request-token --audience "$BUILDKITE_PLUGIN_GCP_WORKLOAD_IDENTITY_FEDERATION_AUDIENCE" --lifetime "${BUILDKITE_PLUGIN_GCP_WORKLOAD_IDENTITY_FEDERATION_LIFETIME:-0}" > "$TMPDIR"/token.json
+buildkite-agent oidc request-token \
+  --audience "$BUILDKITE_PLUGIN_GCP_WORKLOAD_IDENTITY_FEDERATION_AUDIENCE" \
+  --lifetime "${BUILDKITE_PLUGIN_GCP_WORKLOAD_IDENTITY_FEDERATION_LIFETIME:-0}" \
+  "${oidc_params[@]}" > "$TMPDIR"/token.json
```

Comment:
if the `oidc_params` array is empty, some bash versions will cause an error when using `oidc_parms[@]`, there are two possible solutions, in my order of preference:
1- ensure the array is never empty, I would do this by creating a `request_options` with the other two that are mandatory
2- use the only (and ugly) alternative that is supported across all bash versions: `${oidc_params[@]+"${oidc_params[@]}"}`

Reply:
Thanks for the suggestions! I will fix up my pr.

Reply:
Agree that 1 is the cleanest option. I also appended the required arguments into `args` so there should not be a situation where the variable is empty.